server/src/test/java/org/opensearch/indices/replication/SegmentReplicationTargetServiceTests.java

```diff
@@ -0,0 +1,149 @@
+        ReplicationCheckpoint checkpoint = new ReplicationCheckpoint(indexShard.shardId(), 0L, 0L, 0L, 0L);
+        SegmentReplicationSourceFactory replicationSourceFactory = mock(SegmentReplicationSourceFactory.class);
+        SegmentReplicationSource replicationSource = mock(SegmentReplicationSource.class);
+
+        SegmentReplicationTargetService sut = new SegmentReplicationTargetService(
+            threadPool,
+            recoverySettings,
+            transportService,
+            replicationSourceFactory
+        );
+
+        final OpenSearchException expectedError = new OpenSearchException("Fail");
+        final SegmentReplicationTarget target = new SegmentReplicationTarget(
+            checkpoint,
+            indexShard,
+            replicationSource,
+            new SegmentReplicationTargetService.SegmentReplicationListener() {
+                @Override
+                public void onReplicationDone(SegmentReplicationState state) {
+                    Assert.fail();
+                }
+
+                @Override
+                public void onReplicationFailure(SegmentReplicationState state, OpenSearchException e, boolean sendShardFailure) {
+                    assertEquals(SegmentReplicationState.Stage.INIT, state.getStage());
+                    assertEquals(expectedError, e.getCause());
+                    assertTrue(sendShardFailure);
+                }
+            }
+        );
+        final SegmentReplicationTarget spy = Mockito.spy(target);
+        doAnswer(invocation -> {
+            final ActionListener<Void> listener = invocation.getArgument(0);
+            listener.onFailure(expectedError);
+            return null;
+        }).when(spy).startReplication(any());
+        sut.startReplication(spy);
+        closeShards(indexShard);
+    }
+
+    public void testBeforeIndexShardClosed_CancelsOngoingReplications() throws IOException {
+        Settings settings = Settings.builder().put("node.name", SegmentReplicationTargetServiceTests.class.getSimpleName()).build();
+        final ClusterSettings clusterSettings = new ClusterSettings(settings, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        final RecoverySettings recoverySettings = new RecoverySettings(settings, clusterSettings);
+        final TransportService transportService = mock(TransportService.class);
+        final IndexShard indexShard = newShard(false, settings);
+        ReplicationCheckpoint checkpoint = new ReplicationCheckpoint(indexShard.shardId(), 0L, 0L, 0L, 0L);
+        SegmentReplicationSourceFactory replicationSourceFactory = mock(SegmentReplicationSourceFactory.class);
+        SegmentReplicationSource replicationSource = mock(SegmentReplicationSource.class);
+
+        SegmentReplicationTargetService sut = new SegmentReplicationTargetService(
+            threadPool,
+            recoverySettings,
+            transportService,
+            replicationSourceFactory
+        );
```

Comment:
This setup seems common to all tests. Consider moving them to a `@Before` setup method